src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,115 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+
+    private static final int INITIAL_CAPACITY = 10;
+    private static final double COEFFICIENT = 1.5;
+
+    private T[] array = (T[]) new Object[INITIAL_CAPACITY];
+    private int size = 0;
+
     @Override
     public void add(T value) {
-
+        ensureCapacity();
+        array[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkIndexBounds(index < 0);
```

Comment:
The checkIndexBounds method is intended to verify that the index is within the bounds of the array size. However, passing 'index < 0' directly to the method is not the best approach. Instead, you should pass the index and check its bounds within the method. Additionally, the condition should also ensure that the index is not greater than the current size, as it's allowed to add at the index equal to size (one past the last element).